### PR TITLE
Switch to TypeScript

### PR DESCRIPTION
**Pull request by fenwikk**
Added TypeScript and annotations for a better developer experience for those who like to have everything typed. "frappe-charts" isn't typed either so most of the types in the `.d.ts` are option types for that package.

Closes #57

### CODE REVIEW

**Comment by Peter-Barrett** [comment]
@himynameisdave it would be great if you could merge this in. Thanks.

**Comment by silvestrevivo** [comment]
Any updates about this?

**Comment by Peter-Barrett** [comment]
For the moment adding `// @ts-nocheck` above the import is doing the job for me but it would be great to have this merged @himynameisdave .

File: package.json [hunk 1]
```diff
@@ -11,19 +11,23 @@
   },
   "dependencies": {
     "frappe-charts": "^1.6.2",
-    "svelte": "^3.48.0"
+    "svelte": "^3.0.0"
```
Comment (himynameisdave) [inline package.json:14]: Why the revert back to `3.0.0` here?

File: package.json [hunk 2]
```diff
@@ -11,19 +11,23 @@
...
     "@testing-library/svelte": "^3.1.1",
+    "@tsconfig/svelte": "^3.0.0",
     "jest": "^28.0.3",
     "jest-environment-jsdom": "^28.0.2",
     "jest-transform-svelte": "^2.1.1",
     "np": "^7.6.1",
     "rollup": "^2.71.1",
-    "rollup-plugin-svelte": "^7.1.0"
+    "rollup-plugin-svelte": "^7.1.0",
+    "tslib": "^2.4.0",
```
Comment (himynameisdave) [inline package.json:29]: Why do we need this?

File: src/components/base.svelte [hunk 3]
```diff
@@ -1,42 +1,35 @@
...
-  export let barOptions = {};
-  export let lineOptions = {};
-  export let tooltipOptions = {};
-  export let colors = [];
-  export let valuesOverPoints = 0;
-  export let isNavigable = false;
-  export let maxSlices = 3;
+  export let data: Data;
+  export let title: string = '';
+  export let type: ChartType = 'line';
```
Comment (himynameisdave) [inline src/components/base.svelte:10]: [nitpick] I get that these types are ambient, but generally I prefer to import and use types directly ([using type imports](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export)). Not sure if this is standard/best practice though, thoughts?

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{
+  "extends": "@tsconfig/svelte/tsconfig.json",
+  "include": ["src/**/*", "**/*.config.js"],
+  "exclude": ["node_modules/*"],
+  "compilerOptions": {
+    "outDir": "lib",
+    "declaration": true,
```
Comment (himynameisdave) [inline tsconfig.json:7]: 👍

**Review by himynameisdave — APPROVED** [review]
@fenwikk thanks for writing these definitions. 🎉 💯 I have a few questions around the added dependencies, but otherwise this looks good to me!

**Comment by himynameisdave** [comment]
Ping @fenwikk, let me know if there's anything I can do to help get this over the finish line.

**Comment by himynameisdave** [comment]
Ping @fenwikk, sorry to be annoying here (as I really appreciate the work you've already done here), but if you're unable to address the comments I'd be happy to wrap things up here, just let me know 🙂

**Comment by SemmelJochen** [comment]
any news on this? Can I help somehow?

File: package.json (hunk 2 above)
Comment (SemmelJochen) [reply package.json:29]: This is a runtime library for [TypeScript](https://www.typescriptlang.org/) that contains all of the TypeScript helper functions.

File: package.json (hunk 1 above)
Comment (fenwikk) [reply package.json:14]: that was a mistake i think

File: src/components/base.svelte (hunk 3 above)
Comment (fenwikk) [reply src/components/base.svelte:10]: doesnt matter to me. could change it to a non global declaration file if you want

**Comment by fenwikk** [comment]
sorry guys just been busy ill fix whatever needs fixing

**Pull request closed without merging**